test/optional_ref_test.cpp

```diff
@@ -0,0 +1,34 @@
+#include "fixed_containers/optional_ref.hpp"
+
+#include <gtest/gtest.h>
+
+namespace fixed_containers
+{
+
+/*
+*/
+TEST(OptionalRef, FixedContainerProperties)
+{
+    using T = OptionalRef<int&>;
```

Comment:
Discussion:
Should the reference be explicit as in `OptionalRef<int&>` or should it get auto-applied as in `OptionalRef<int>`

Reply:
would one point in favor of `OptionalRef<int>` be that CTAD would work correctly? we could do something like `OptionalRef maybe_foo(my_int)`, where as with the explicit one, this experssion would need to be `OptionalRef maybe_foo(std::ref(my_int))`. is this right?

Reply:
We should try to emulate existing STL APIs if possible. `std::reference_wrapper<T>` is probably the closest analog.

The code example in the documentation expresses the template argument in terms of the base type `T` and not `T&`.

```cpp
std::vector<std::reference_wrapper<int>> v(l.begin(), l.end());
```

Reply:
Familiarity with `std::reference_wrapper<T>` is a pretty good reason to do `OptionalRef<int>`.
Additionally, I tried using it with the `OptionalRef<int&>` form and indeed CTAD is problematic, and `const int&` vs `int&` is awkward as well. Also, there are paper-cuts when trying to use a `T` with implied ref, the existing comment of "using ref  = T; // T is required to be a reference type" attests to that as well.

I vote for `OptionalRef<int>` too.

Reply:
alright, will go ahead with `OptionalRef<int>`